src/common/object.ts

```diff
@@ -40,3 +40,25 @@ export const nameRegex = new RegExp(/^(.){1,60}$/);
 export const socialHandleRegex = new RegExp(/^@?([\w-]){1,39}$/i);
 export const handleRegex = new RegExp(/^@?[a-z0-9](\w){2,38}$/i);
 export const descriptionRegex = new RegExp(/^[\S\s]{1,250}$/);
+
+export const setDefaultsIfNull = <T extends object>(obj: T, defaults: T): T => {
+  if (isNullOrUndefined(obj)) {
+    return defaults;
+  }
+
+  if (isNullOrUndefined(defaults)) {
+    return obj;
+  }
+
+  return Object.keys(obj).reduce((result, key) => {
+    const value = obj[key];
+
+    if (isNullOrUndefined(value)) {
+      return result;
+    }
+
+    result[key] = value;
+
+    return result;
+  }, defaults);
```

Comment:
While I get what you are trying to do this seems too complicated in my opinion. 
- it is doing iteration (over keys) which just adds more computation on field resolver using it
- what if someone introduces nested object, this would not be able to resolve it? 

I personally much more prefer just doing it inline in gql as you did before https://github.com/dailydotdev/daily-api/pull/1857/commits/e31fd547bf5bc65c2e4e4b152a8e6295860d2e5c#diff-95123b05e21b34d3705f1b3748fe295fa17db5caeeb22238a294368729175a07R1649-R1652

Reply:
@rebelchris @DragosIorgulescu tagging for your opinion as well.

Reply:
Yeah, I just realized if we add another property, we might miss another `obj.key ?? value` to put since everything is inline. For now, yeah, this won't cover a nested object, but that can definitely add that processing once we get the need for it.

Reply:
> Yeah, I just realized if we add another property, we might miss another obj.key ?? value 

Yeah, figured, did you look into option to take care of this with Typescript instead by defining return type of field resolver, which would be caught in lint/test before any deployment. So it would make sure additional fields do not miss handling without introducing runtime cost.

Reply:
Yeah, I checked, and it could be possible, just needs more work on the GraphORM side since it doesn't automatically understand the type it needs to return.

Reply:
I tested with:
```ts
transform: (value: SourceFlagsPublic): SourceFlagsPublic => {
  return {
    totalPosts: value?.totalPosts ?? 0,
    totalViews: value?.totalViews ?? 0,
    totalUpvotes: value?.totalUpvotes ?? 0,
    featured: value?.featured ?? false,
  };
}
```

and then adding new field `test` to `SourceFlagsPublic`

I automatically get the error:
<img width="834" alt="image" src="https://github.com/dailydotdev/daily-api/assets/9803078/ec7cf279-20a2-4b16-8779-e2ea05325444">

Which would block any pipeline and force developer to add handling in graphorm.

Reply:
Right. Yeah, we can probably do that instead of adding generic types to GraphORM.

Reply:
Ideally, we can do generic types for better devex and avoid forgetting to add that type on the function since our lint does not ask for it.

Reply:
Yeah, problem partially is that we don't have [strict option in typescript](https://dailydotdev.atlassian.net/browse/WT-1532), it would light up red all over the place even more during dev.